### PR TITLE
refactor: Apply all typing changes in medmodels/matching according to pyright

### PR DESCRIPTION
**Pull request by MarIniOnz**

### CODE REVIEW

File: medmodels/matching/metrics.py [hunk 1]
```diff
@@ -1,19 +1,20 @@
-from typing import Tuple
+"""Metrics for comparing vectors in the context of matching classes."""
 
 import numpy as np
+from numpy.typing import NDArray
 
 
-def absolute_metric(*vectors: Tuple[np.ndarray, np.ndarray]) -> float:
+def absolute_metric(*vectors: NDArray[np.float64]) -> float:
```
Comment (LauraBoenchenLB) [inline medmodels/matching/metrics.py:7]: Can we still use the * here if the input is a NDArray and not a Tuple?

File: medmodels/matching/metrics.py [hunk 2]
```diff
@@ -26,7 +27,7 @@ def absolute_metric(*vectors: Tuple[np.ndarray, np.ndarray]) -> float:
     return sum(np.abs(diff))
 
 
-def exact_metric(*vectors: Tuple[np.ndarray, np.ndarray]) -> float:
+def exact_metric(*vectors: NDArray[np.float64]) -> float:
```
Comment (LauraBoenchenLB) [inline medmodels/matching/metrics.py:30]: Same question

**Review by LauraBoenchenLB — COMMENTED** [review]
Just one question where I'm not sure if the use of the argument is right. Otherwise great PR, the typing looks great and tests also match!

File: medmodels/matching/metrics.py (hunk 2 above)
Comment (MarIniOnz) [reply medmodels/matching/metrics.py]: Same.

**Review by JabobKrauskopf — APPROVED** [review]
Nice job! :)

Minor adjustments and please add `matching` to the pyright config in `pyproject.toml` (in this PR)

**Review by JabobKrauskopf — CHANGES_REQUESTED** [review]
sry i meant to request changes

File: medmodels/matching/covariates/covariates_preprocessing.py [hunk 3]
```diff
@@ -30,22 +37,26 @@ def covariate_coarsen(covariate: np.ndarray, n_bins: int = 10) -> np.ndarray:
     return np.digitize(covariate, bins)
 
 
-def covariate_add_noise(covariate: Series, n_digits: int = 2) -> Series:
+def covariate_add_noise(
+    covariate: pd.Series[Union[int, float]], n_digits: int = 2
```
Comment (JabobKrauskopf) [inline medmodels/matching/covariates/covariates_preprocessing.py:41]: Should `n_digits` be `n_digits: Optional[int] = 2`?

**Review by JabobKrauskopf — APPROVED** [review]
Good job!

**Review by LauraBoenchenLB — APPROVED** [review]

**Pull request merged**